### ISSUE TITLE
Argument 2 passed to OC_App::isAppCompatible() must be of the type array, null given

### ISSUE THREAD

**Issue by jancborchardt** [nextcloud/server#8964]
On a recent update to master, I ran into this Internal Server Error and can not recover from it:

```
Technical details

    Remote Address: 127.0.0.1
    Request ID: 2Rn8ViN3uHj0d64lFOhz
    Type: TypeError
    Code: 0
    Message: Argument 2 passed to OC_App::isAppCompatible() must be of the type array, null given, called in /home/jan/nextcloud/lib/private/App/AppManager.php on line 414
    File: /home/jan/nextcloud/lib/private/legacy/app.php
    Line: 826


Trace

#0 /home/jan/nextcloud/lib/private/App/AppManager.php(414): OC_App::isAppCompatible('14.0.0.1', NULL)
#1 /home/jan/nextcloud/lib/base.php(377): OC\App\AppManager->getIncompatibleApps('14.0.0.1')
#2 /home/jan/nextcloud/lib/base.php(936): OC::printUpgradePage(Object(OC\SystemConfig))
#3 /home/jan/nextcloud/index.php(37): OC::handleRequest()
#4 {main}
```

Stable13 works without any problem. By now I already went and disabled every single app via the app management, but that does not solve the problem either.

Any ideas?

**Comment by juliushaertl** [comment]
This is caused by the type hinting added by 9dc3b0480228a3c35b3b7a7cb1a904006f7e0762

@jancborchardt My guess would be that there is an app that doesn't return proper app info, can you check by adding the following snippet after https://github.com/nextcloud/server/blob/514de5dfa18d92d2f7bb15c9840d685b25300f4c/lib/private/App/AppManager.php#L413

```
if ($info === null) {
  \OC::$server->getLogger()->error('$info null for ' . $appId);
}
```

I'll prepare a patch to not fail hard in that case, but we should also investigate the root cause.

**Comment by jancborchardt** [comment]
@juliushaertl thanks! It seems to have been `nextcloud_announcements`:
```
{"reqId":"GoEHI24hlsLXdoIZZfEr","level":3,"time":"2018-03-27T12:03:47+00:00","remoteAddr":"127.0.0.1","user":"--","app":"no app in context","method":"GET","url":"\/nextcloud\/","message":"$info null for nextcloud_announcements","userAgent":"Mozilla\/5.0 (X11; Ubuntu; Linux x86_64; rv:59.0) Gecko\/20100101 Firefox\/59.0","version":"13.0.1.1"}
{"reqId":"GoEHI24hlsLXdoIZZfEr","level":3,"time":"2018-03-27T12:03:47+00:00","remoteAddr":"127.0.0.1","user":"--","app":"index","method":"GET","url":"\/nextcloud\/","message":"Exception: {\"Exception\":\"TypeError\",\"Message\":\"Argument 2 passed to OC_App::isAppCompatible() must be of the type array, null given, called in \\\/home\\\/jan\\\/nextcloud\\\/lib\\\/private\\\/App\\\/AppManager.php on line 417\",\"Code\":0,\"Trace\":\"#0 \\\/home\\\/jan\\\/nextcloud\\\/lib\\\/private\\\/App\\\/AppManager.php(417): OC_App::isAppCompatible('14.0.0.1', NULL)\\n#1 \\\/home\\\/jan\\\/nextcloud\\\/lib\\\/base.php(377): OC\\\\App\\\\AppManager->getIncompatibleApps('14.0.0.1')\\n#2 \\\/home\\\/jan\\\/nextcloud\\\/lib\\\/base.php(936): OC::printUpgradePage(Object(OC\\\\SystemConfig))\\n#3 \\\/home\\\/jan\\\/nextcloud\\\/index.php(37): OC::handleRequest()\\n#4 {main}\",\"File\":\"\\\/home\\\/jan\\\/nextcloud\\\/lib\\\/private\\\/legacy\\\/app.php\",\"Line\":826}","userAgent":"Mozilla\/5.0 (X11; Ubuntu; Linux x86_64; rv:59.0) Gecko\/20100101 Firefox\/59.0","version":"13.0.1.1"}
```

I think it was because I removed some apps before update (among them nextcloud_announcements) because of this warning about updating apps to the proper version etc. And now that I checked, nextcloud_announcements was just an empty folder.
Removing the empty folder did not help, only recloning from https://github.com/nextcloud/nextcloud_announcements/

– Aaand now it’s password_policy:
```
{"reqId":"KHQ1FEJP8O0akgw1Z233","level":3,"time":"2018-03-27T12:07:52+00:00","remoteAddr":"127.0.0.1","user":"--","app":"no app in context","method":"GET","url":"\/nextcloud\/","message":"$info null for password_policy","userAgent":"Mozilla\/5.0 (X11; Ubuntu; Linux x86_64; rv:59.0) Gecko\/20100101 Firefox\/59.0","version":"13.0.1.1"}
```

I assume it was wrong to remove those apps? Are they shipped by default or so? Will update as I fix apps step by step, but this feels like it shouldn’t fail hard even if the apps are removed?

**Comment by jancborchardt** [comment]
Those two were actually it! Thanks a lot @juliushaertl, now the update went through. :tada:

**Comment by juliushaertl** [comment]
Fix is in https://github.com/nextcloud/server/pull/9008

**Comment by MorrisJobke** [comment]
Fixed with #9008
